float32 current
---
bool success
